### ISSUE TITLE
Options should support more function spaces

### ISSUE THREAD

**Issue by tkarna** [thetisproject/thetis#25]
Currently function spaces are chosen with the boolean `options.mimetic`: `False` corresponds to P(n)DG-P(n)DG space and `True` to RT(n+1)-P(n)DG pair.

#11 adds an additional `options.continuous_pressure` that provides P(n)DG-P(n+1) pair.
All cases use `options.order` as the base degree (n).

This needs to be generalized to support more function spaces. Easy way is to define names for each combination, for example

```
options.element_family = 'dg-dg'  # p(n)dg-p(n)dg
options.element_family = 'rt-dg'  # rt(n+1)dg-p(n)dg
options.element_family = 'dg-cg'  # p(n)dg-p(n+1)
```

The above names do not contain the degrees as sane combinations are presumably dictated by the stability constraints of shallow water equations. (Are there exceptions?)
Alternatively we could use human readable names (if such exist)?

**Comment by tkarna** [comment]
While changing this we could replace `options.order` by `options.degree`.

**Comment by tkarna** [comment]
In addition P2-P1 pair could be called `'taylor-hood'`.

**Comment by tkarna** [comment]
This is now implemented in #54.
